### ISSUE TITLE
Feature requests

### ISSUE THREAD

**Issue by chaoscreater** [HenriWahl/Nagstamon#484]
First of all, thanks for developing this awesome tool. I have been using it for a while now and it has made my job as a sysadmin much easier.

I do have a few suggestions and was wondering if you are able to implement them? We're mainly using Nagios Core 4.0.8 and CheckMK 1.2.8p18.

1. "Hosts and services with disabled checks" filters out both active and passive checks. There are some hosts / services where we use passive checks and not active checks. When passive check is in use, active checks will be disabled and this gets filtered out by Nagstamon. Why the need for passive checks? Well, there are multiple reasons, see here for more info:
http://nagios.manubulon.com/traduction/docs25en/passivechecks.html

It would be ideal if we can be a bit more granular with this filter. Maybe add someting like "Hosts and services with disabled Passive checks" and "Hosts and services with disabled Active checks".

2. Select multiple entries and batch action them. For example, there may be multiple servers that are low on drive space. Instead of acknowledging each one and assigning each alert to an engineer to work on, we can batch acknowledge them and assign it to the same person in one go. There will need to be checkboxes basically, or support CTRL + mouse click to select multiple entries.

3. Multi-sort support. Kind of like Excel, where you can sort on multiple columns, it'd be nice if you can do the same in Nagstamon. I would like to sort by the Duration, then sort by e.g. Status or maybe sort by whether the service is flapping, whether it is a passive check, whether it has a comment, etc. 

4. In Nagios, you can look at "Tactical Monitoring Overview". This is what we see:

![image](https://user-images.githubusercontent.com/18227319/37958480-214ce452-320d-11e8-83ea-20dd3d6bd2da.png)

In Nagstamon, I would like to modify these buttons (there isn't a setting to modify them):

![image](https://user-images.githubusercontent.com/18227319/37958486-2632953e-320d-11e8-93f3-0bed501e535e.png)

For Monitor, I would like to edit the URL to point to the page that shows the Tactical Monitoring Overview, rather than just going to the Nagios homepage. For hosts and services, I want to edit the button URL to take me to Unhandled Problems. I don't care about services that are showing Warning. The Services button currently just takes you to the page that shows all Services, both warning and critical.

5. Filter out host and/or services with comments and also display the comment icon in Nagstamon. If Nagstamon can show icons for e.g. passive checks or flapping services, why not also show the comment icon too? It would be really awesome if we can filter anything that has a comment.

![image](https://user-images.githubusercontent.com/18227319/37958499-2ee48214-320d-11e8-8711-c81ece2f2fb5.png)

For CheckMK, you can edit views and show host and service comments:

![image](https://user-images.githubusercontent.com/18227319/37958510-36ad8310-320d-11e8-8b9e-c3c0443e627a.png)

**Comment by HenriWahl** [comment]
Hi - sorry for the late comment, but there is so much other stuff to do.
Well, reworking the filters is a long planned change which still waits to be implemented.
Same applies to customizable buttons.
Comments are not transmitted to Nagstamon so they would have to be requested for one for one - which is a good strategy for a DOS attack...

**Comment by chaoscreater** [comment]
Hi Henri,

Regarding the "Comments" suggestion, what if we create a dashboard view in CheckMK to exclude any services/hosts with a comment and then add the dashboard to Nagstamon? That way, you don't really have to re-work the comment feature. The problem is that I can't simply add a dashboard view to Nagstamon. It doesn't like it.

I was hoping to modify it myself (I have very limited knowledge in programming). I'm getting the following error when I try to compile your source code:

>     VERSION = AppInfo.VERSION.replace('-', '.') + '.' + DIST + DIST_VERSION
> NameError: name 'DIST' is not defined

**Comment by HenriWahl** [comment]
I am not sure if this would work because a host/service might have several comments.

**Comment by chaoscreater** [comment]
That's fine, Even if there is just a comment icon in the column (just like how Nagios supports it in the screenshot I posted above), then that will be extremely helpful for us.

**Comment by HenriWahl** [comment]
At least the second wish became true now - please check lastest unbstable version for multiple item selection.

**Comment by chaoscreater** [comment]
I managed to get the Comment feature implemented myself (sort of). It's actually super easy to implement and this is coming from someone who isn't good at programming and has no idea how to program in Python. 

I don't know how to add an extra column in Nagstamon, so I ended up just taking an existing column that I don't find useful, then added a few lines to show that the alert has a Comment against it.

Basically, you take the html code of the Comment icon shown in Nagios:

![image](https://user-images.githubusercontent.com/18227319/52896376-d466cc00-322b-11e9-9fd6-58e9910ddc97.png)

Then you simply output either True/False to show if the alert has a comment against it. This doesn't show you the actual comment, but it is good enough in my use case. If there is a comment for an alert, it'll show "True" in the column against the alert. Why don't I simply acknowledge the alert? Because in our use case, we rotate weekly to check Nagios (and other dashboards) and assign out alerts to agents that actually deal with them. Sometimes, something can't be resolved easily or quickly and therefore the agent will leave it unassigned on Nagios and just monitor and slowly work at resolving it. If the alert were to be acknowledged and the agent is away, nobody will be notified and they won't be able to quickly see that on the Nagios dashboard. 

Therefore, as someone who assigns out these alerts, we may see a bunch of them that can't be acknowledged, but has a comment against it to show that it is assigned to person XYZ. If we know something has a comment, we can skip them and not assign them out again.
